File: backend/app/schemas/ai.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


AiInsightStatus = Literal["active", "accepted", "rejected", "archived"]


class AiInsightResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    related_goal_id: int | None

    insight_date: date

    insight_type: str
    source: str

    title: str
    content: str
    recommendation: str | None

    status: str

    created_at: datetime
    updated_at: datetime


class AiInsightStatusUpdate(BaseModel):
    status: AiInsightStatus

File: frontend/src/types/ai.ts
export type AiInsightStatus = "active" | "accepted" | "rejected" | "archived";

export type AiInsightResponse = {
  id: number;
  user_id: number;
  related_goal_id: number | null;

  insight_date: string;

  insight_type: string;
  source: string;

  title: string;
  content: string;
  recommendation: string | null;

  status: string;

  created_at: string;
  updated_at: string;
};

export type AiInsightStatusUpdate = {
  status: AiInsightStatus;
};